src/trusted/desc/nacl_desc_io.c

```diff
@@ -63,20 +63,14 @@ int NaClDescIoDescCtor(struct NaClDescIoDesc  *self,
   struct NaClDesc *basep = (struct NaClDesc *) self;
   int rv;
 
-  if (!basep) {
-    return -NACL_ABI_EBADF;
-  }
-  basep->base.vtbl = NULL;
+  basep->base.vtbl = (struct NaClRefCountVtbl const *) NULL;
   if (!NaClDescCtor(basep)) {
     return 0;
   }
   rv = NaClDescIoDescSubclassCtor(self, hd);
   if (!rv) {
     (*NACL_VTBL(NaClRefCount, basep)->Dtor)((struct NaClRefCount *) basep);
   }
-  if (hd) {
-    NACL_VTBL(NaClDesc, basep)->SetFlags(basep, hd->flags & NACL_ABI_O_ACCMODE);
-  }
```

Comment:
Why can we get rid of this?

Reply:
Hmm. I just reverted this back to its pre lind-hacked up state. I'm not sure why this got removed.

Reply:
Okay. Joey added this sometime in between while adding some syscalls it seems. Removing this shouldn't interfere with any functionality since that stuff was all removed as well.